src/LayerPlan.cpp

```diff
@@ -774,6 +774,31 @@ void LayerPlan::writeGCode(GCodeExport& gcode)
                 }
                 gcode.writeTemperatureCommand(prev_extruder, prev_extruder_temp, wait);
             }
+
+            if (!getPrimeTowerIsPlanned())
+            {
+                // when spiralizing, there should be no prime tower but to avoid huge blobs occuring at extruder changes
+                // we do the post extruder swap prime into thin air (before extruder is moved back to the print)
+                bool first_extrusion_move_is_spiralized = false;
+                std::vector<GCodePath>& paths = extruder_plan.paths;
+                for(unsigned int path_idx = 0; path_idx < paths.size(); path_idx++)
+                {
+                    if (!paths[path_idx].config->isTravelPath())
+                    {
+                        first_extrusion_move_is_spiralized = paths[path_idx].spiralize;
+                        break;
+                    }
+                }
+                if (first_extrusion_move_is_spiralized)
+                {
+                    // set the extrusion temperature and wait for it to be achieved
+                    gcode.writeTemperatureCommand(extruder, extruder_plan.extrusion_temperature.value_or(extruder_plan.required_start_temperature), true);
```

Comment:
I think you misread the gcode you generated.
What I see in your generated gcode is that it requires the initial_printing_temp right after switching the extruder and then does an M104 to start heating to the printing_temp without waiting for it to be reached.

Again: if there is a problem with temperature control then that problem should be fixed in the temperature control code. It shouldn't be fixed by introducing a hotfix for spiralize only.